### DISCUSSION TITLE
nan/inf in mysql handler

### DISCUSSION THREAD

**Discussion by youngsofun** [databendlabs/databend#8941] [General]
https://dev.mysql.com/doc/refman/5.7/en/problems-with-float.html
it`s `nan` here

python driver seems to like "nan"? @ariesdevil 

mysql  client display "NULL"

```
mysql> select 1/0
    -> ;
+------+
| 1/0  |
+------+
| NULL |
```

mysql jdbc not accept 'inf': 
https://github.com/datafuselabs/databend/pull/8929#issuecomment-1325856898

**Comment by youngsofun** [comment]
cc @sandflee

**Comment by sundy-li** [comment]
mysql has no inf or nan for float value, it can only accepts number or null.

**Reply by youngsofun** [reply]
ok, so we should output NULL (without quote)?

**Reply by youngsofun** [reply]
and do not need to consider  "NULL" as valid input for ` float  not null`?

**Comment by youngsofun** [comment]
mysql jdbc accept NaN
https://github.com/mysql/mysql-connector-j/blob/27603148f10a5f47467bec7ad26a5ca28da63c72/src/main/protocol-impl/java/com/mysql/cj/protocol/a/MysqlTextValueDecoder.java#L248

https://stackoverflow.com/questions/10750975/parse-string-to-double-nan

**Reply by FANNG1** [reply]
I changed the const value `NAN_BYTES_LOWER` to `NaN` and `INF_BYTES_LOWER` to `Infinity` in databend,    mysql jdbc could query out the data.

**Reply by youngsofun** [reply]
what if we change them to NULL both?

**Reply by youngsofun** [reply]
https://docs.oracle.com/javase/7/docs/api/java/lang/Double.html#valueOf%28java.lang.String%29

**Reply by youngsofun** [reply]
because MySQL client seems to print what the server sent  it and it print NULL

**Reply by FANNG1** [reply]
yes, `null` is reasonable.   but in our env, we prefer to keep consistent with hive-jdbc. how about output `Nan` `Infinity `  if `sql_dialect` is `Hive` , otherwise output NULL ?

**Reply by youngsofun** [reply]
python, java , and golang is mostly used, 
They all support "NaN" & "Infinity"
we can just use them for MySQL?

**Reply by youngsofun** [reply]
cc @sundy-li

**Comment by youngsofun** [comment]
@sundy-li  @ariesdevil 

in logic_test, MySQL connector convert string to python float then we print the python float

```
(venv) ➜  python 
Type "help", "copyright", "credits" or "license" for more information.
>>> float("1.1")
1.1
>>> float("nan")
nan
>>> float("NaN")
nan
>>> float("Inf")
inf
>>> float("infinity")
inf
>>> float("infinity")
```

**Reply by sundy-li** [reply]
So we can wrap the result to be NaN or disable the tests

**Comment by youngsofun** [comment]
like python, golang ParseFloat support multi form toohttps://pkg.go.dev/strconv#ParseFloat

**Reply by youngsofun** [reply]
https://pkg.go.dev/strconv#ParseFloat
